### PR TITLE
bpo-40995: reprlib.Repr attributes can be overriden in __init__()

### PR DESCRIPTION
**Pull request by remilapeyre**
https://bugs.python.org/issue40995

### CODE REVIEW

File: Doc/library/reprlib.rst [hunk 1]
```diff
@@ -17,7 +17,9 @@ debugger and may be useful in other contexts as well.
 This module provides a class, an instance, and a function:
 
 
-.. class:: Repr()
+.. class:: Repr(self, *, maxlevel=6, maxtuple=6, maxlist=6, maxarray=5, \
```
Comment (ZackerySpytz) [inline Doc/library/reprlib.rst:20]: `self` should not be included here.

File: Doc/library/reprlib.rst [hunk 2]
```diff
@@ -71,72 +73,75 @@ string instead.
 Repr Objects
 ------------
 
-:class:`Repr` instances provide several attributes which can be used to provide
-size limits for the representations of different object types,  and methods
-which format specific object types.
+.. class:: Repr(self, *, maxlevel=6, maxtuple=6, maxlist=6, maxarray=5, \
```
Comment (ZackerySpytz) [inline Doc/library/reprlib.rst:76]: `self` should not be included here.

File: Doc/library/reprlib.rst [hunk 3]
```diff
@@ -17,7 +17,9 @@ debugger and may be useful in other contexts as well.
 This module provides a class, an instance, and a function:
 
 
-.. class:: Repr()
+.. class:: Repr(self, *, maxlevel=6, maxtuple=6, maxlist=6, maxarray=5, \
+                 maxdict=4, maxset=6, maxfrozenset=6, maxdeque=6, maxstring=30, \
+                 maxlong=40, maxother=30)
```
Comment (ZackerySpytz) [inline Doc/library/reprlib.rst:22]: Please align these lines with the `(`.

File: Doc/library/reprlib.rst [hunk 4]
```diff
@@ -71,72 +73,75 @@ string instead.
 Repr Objects
 ------------
 
-:class:`Repr` instances provide several attributes which can be used to provide
-size limits for the representations of different object types,  and methods
-which format specific object types.
+.. class:: Repr(self, *, maxlevel=6, maxtuple=6, maxlist=6, maxarray=5, \
+                 maxdict=4, maxset=6, maxfrozenset=6, maxdeque=6, maxstring=30, \
+                 maxlong=40, maxother=30)
```
Comment (ZackerySpytz) [inline Doc/library/reprlib.rst:78]: Please align these lines with the `(`.

File: Lib/test/test_reprlib.py [hunk 5]
```diff
@@ -216,6 +216,13 @@ def test_unsortable(self):
         r(y)
         r(z)
 
+    def test___init__(self):
+        a = "a"*40
+        r = Repr()
+        self.assertEqual(r.repr(a), "'aaaaaaaaaaaa...aaaaaaaaaaaaa'")
+        r = Repr(maxstring=10)
+        self.assertEqual(r.repr(a), "'aa...aaa'")
```
Comment (ZackerySpytz) [inline Lib/test/test_reprlib.py:224]: I think more tests need to be provided.

File: Doc/library/reprlib.rst [hunk 6]
```diff
@@ -71,72 +73,75 @@ string instead.
...
 
+   :class:`Repr` instances provide several attributes which can be used to provide
```
Comment (ZackerySpytz) [inline Doc/library/reprlib.rst:80]: Please limit lines to 80 characters in reST files (https://devguide.python.org/documenting/#use-of-whitespace).

**Review by ZackerySpytz — COMMENTED** [review]
I'm not sure if this patch will be accepted, but I have some comments.

**Comment by gvanrossum** [comment]
Looks like some version of this was actually merged, so closing.

**Pull request closed without merging**